### ISSUE TITLE
Data utilization issue

### ISSUE THREAD

**Issue by ATZQIN** [chu0802/SLA#18]
According to the code as below, we can know that you use the unlabeled testset in target domain to calculate the class center, but in the Figure 3 in your paper, unlabeled trainset should be the input data of the protonet with pseudo center. I am looking forward to your reply. Thank you very much. 
def ppc_update(self, step):
    if step == self.config.warmup:
        self.ppc.init(self.model, self.loaders.loaders["target_unlabeled_test"])
        self.lr_scheduler.refresh()

    if step > self.config.warmup and step % self.config.update_interval == 0:
        self.ppc.init(self.model, self.loaders.loaders["target_unlabeled_test"])

**Comment by chu0802** [comment]
Hello,

Thanks for raising the issue.
The confusion might arise from my bad naming conventions. I am sorry for that. 
However, `target_unlabeled_test` and `target_unlabeled_train` actually refer to the same datsaet.

As can be seen in my dataset preparation script ([here](https://github.com/chu0802/SLA/blob/1cc56de39f99bc3d3943332b1ee5550dbba279eb/src/dataset.py#L152)):

```python
    def get_list_file_name(self):
        if self.domain_type == "source":
            return "all.txt"
        if self.mode == "validation":
            return "val.txt"
        if self.labeled == "labeled":
            return f"train_{self.num_shot}.txt"
        return f"test_{self.num_shot}.txt"
```

Both `target_unlabeled_train` and `target_unlabeled_test` refer to the same file `test_{self.num_shot}.txt`

The only difference between these two datasets is the dataloader used.

`target_unlabeled_train` use [InfiniteDataLoader](https://github.com/chu0802/SLA/blob/1cc56de39f99bc3d3943332b1ee5550dbba279eb/src/dataset.py#L98C7-L98C25) instead of the standard data loader (`DataLoader`) to make sure that the data can be loaded infinitely. (Check [here](https://github.com/chu0802/SLA/blob/1cc56de39f99bc3d3943332b1ee5550dbba279eb/src/dataset.py#L174) for more details).

To answer your initial question: we use `target_unlabeled_test` to initialize PPC because we don't need to infinitely sample the target data. (i.e., each ssample should only be used once to calculate pseudo centers)

Hope this helps!

Best,
Yu-Chu

**Comment by ATZQIN** [comment]
Thank you very much for taking the time to answer my questions in detail. Your detailed reply helps me understand your article and code very well.

Wishing you the best of everything.





































At 2025-02-20 15:41:02, "Yu-Chu Yu" ***@***.***> wrote:

Hello,

Thanks for raising the issue.
The confusion might arise from my bad naming conventions. I am sorry for that.
However, target_unlabeled_test and target_unlabeled_train actually refer to the same datsaet.

As can be seen in my dataset preparation script (here):

defget_list_file_name(self):
        ifself.domain_type=="source":
            return"all.txt"ifself.mode=="validation":
            return"val.txt"ifself.labeled=="labeled":
            returnf"train_{self.num_shot}.txt"returnf"test_{self.num_shot}.txt"

Both target_unlabeled_train and target_unlabeled_test refer to the same file test_{self.num_shot}.txt

The only difference between these two datasets is the dataloader used.

target_unlabeled_train use InfiniteDataLoader instead of the standard data loader (DataLoader) to make sure that the data can be loaded infinitely. (Check here for more details).

To answer your initial question: we use target_unlabeled_test to initialize PPC because we don't need to infinitely sample the target data. (i.e., each ssample should only be used once to calculate pseudo centers)

Hope this helps!

Best,
Yu-Chu

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>

chu0802 left a comment (chu0802/SLA#18)

Hello,

Thanks for raising the issue.
The confusion might arise from my bad naming conventions. I am sorry for that.
However, target_unlabeled_test and target_unlabeled_train actually refer to the same datsaet.

As can be seen in my dataset preparation script (here):

defget_list_file_name(self):
        ifself.domain_type=="source":
            return"all.txt"ifself.mode=="validation":
            return"val.txt"ifself.labeled=="labeled":
            returnf"train_{self.num_shot}.txt"returnf"test_{self.num_shot}.txt"

Both target_unlabeled_train and target_unlabeled_test refer to the same file test_{self.num_shot}.txt

The only difference between these two datasets is the dataloader used.

target_unlabeled_train use InfiniteDataLoader instead of the standard data loader (DataLoader) to make sure that the data can be loaded infinitely. (Check here for more details).

To answer your initial question: we use target_unlabeled_test to initialize PPC because we don't need to infinitely sample the target data. (i.e., each ssample should only be used once to calculate pseudo centers)

Hope this helps!

Best,
Yu-Chu
